frontend/src/routes/manage/Shared/EditVideoList.tsx

```diff
@@ -0,0 +1,374 @@
+        </div>
+        {description && <p css={{ marginBottom: 8, maxWidth: 750 }}>{description}</p>}
+        <div css={{ marginBottom: 12, display: "flex", gap: 8 }}>
+            <AddVideoMenu {...{ setEvents, events, floatingRef }} />
+            {/* // Todo: Omit upload button when adding this route for playlists */}
+            <LinkButton to={UploadRoute.urlWithSeries({ seriesId: listId })}>
+                {t("manage.video-list.edit.upload")}
+            </LinkButton>
+        </div>
+        {events.length > 0 && <>
+            <div css={{
+                maxHeight: 360,
+                overflowY: "auto",
+                border: `1px solid ${COLORS.neutral25}`,
+                borderRadius: 8,
+            }}>
+                {events.map(event => <EventEntry key={event.id} {...{
+                    event, events, setEvents,
+                }} />)}
+            </div>
+            <SubmitButtonWithStatus
+                label={t("manage.video-list.edit.save")}
+                onClick={onSubmit}
+                disabled={!!commitError || events.every(e => e.action === "none") || inFlight}
+                {...{ inFlight, success, setSuccess }}
+            />
+        </>}
+        {boxError(commitError)}
+    </Inertable>;
+};
+
+type AddVideoMenuProps = {
+    floatingRef: React.RefObject<FloatingHandle>;
+    events: ListEvent[];
+    setEvents: React.Dispatch<React.SetStateAction<ListEvent[]>>;
+};
+
+const AddVideoMenu: React.FC<AddVideoMenuProps> = ({ events, setEvents, floatingRef }) => {
+    const { t } = useTranslation();
+    const isDark = useColorScheme().scheme === "dark";
+    const [buttonIsActive, setButtonIsActive] = useState(false);
+
+    useOnOutsideClick(floatingRef, () => setButtonIsActive(false));
+
+    return (
+        <FloatingContainer
+            ref={floatingRef}
+            trigger="click"
+            placement="right-start"
+            borderRadius={8}
+            ariaRole="menu"
+            css={{ width: "fit-content" }}
+        >
+            <FloatingTrigger>
+                <Button onClick={() => setButtonIsActive(prev => !prev)} css={{
+                    ...buttonIsActive && { "&&": {
+                        borderColor: COLORS.neutral60,
+                        backgroundColor: COLORS.neutral15,
+                    } },
+                }}>
+                    {t("manage.video-list.edit.add-video")}
+                </Button>
+            </FloatingTrigger>
+            <Floating
+                padding={0}
+                borderWidth={isDark ? 1 : 0}
+                backgroundColor={isDark ? COLORS.neutral15 : COLORS.neutral05}
+                shadowBlur={12}
+                shadowColor="rgba(0, 0, 0, 30%)"
+                css={{
+                    width: 400,
+                    height: 362,
+                    padding: 8,
+                    paddingBottom: 0,
+                }}
+                hideArrowTip
+            >
+                <EventSelector
```

Comment:
In here, event titles are not cut off after N lines, so really long titles look bad. And also: the selector looks fairly narrow, couldn't we make it wider?

Reply:
Yes I think fixing this in this PR is fine